server/src/main/java/org/apache/iotdb/db/qp/executor/PlanExecutor.java

```diff
@@ -653,15 +653,15 @@ private void addRowRecordForShowQuery(
   @Override
   public void delete(DeletePlan deletePlan) throws QueryProcessException {
     try {
-      Set<String> existingPaths = new HashSet<>();
+      Set<List<String>> existingPaths = new HashSet<>();
       for (Path p : deletePlan.getPaths()) {
-        existingPaths.addAll(getPathsName(p.getFullPath()));
+        existingPaths.addAll(getPathsName(p.getNodes()));
```

Comment:
Set<List<String>> can't be used directly.
List<String> a = List.of("a", "b").
List<String> b = List.of ("a", "b").

a is not equal to b.